Source/Core/Core/IOS/ES/NandUtils.cpp

```diff
@@ -174,10 +177,32 @@ std::vector<IOS::ES::Content> ES::GetStoredContentsFromTMD(const IOS::ES::TMDRea
   std::vector<IOS::ES::Content> stored_contents;
 
   std::copy_if(contents.begin(), contents.end(), std::back_inserter(stored_contents),
-               [this, &tmd, &map](const IOS::ES::Content& content) {
+               [this, &tmd, &map, check_content_hashes](const IOS::ES::Content& content) {
+                 const auto fs = m_ios.GetFS();
+
                  const std::string path = GetContentPath(tmd.GetTitleId(), content, map);
-                 return !path.empty() &&
-                        m_ios.GetFS()->GetMetadata(PID_KERNEL, PID_KERNEL, path).Succeeded();
+                 if (path.empty())
+                   return false;
+
+                 // Check whether the content file exists.
+                 const auto file = fs->OpenFile(PID_KERNEL, PID_KERNEL, path, FS::Mode::Read);
+                 if (!file.Succeeded())
+                   return false;
+                 const auto status = file->GetStatus();
+                 if (!status.Succeeded())
```

Comment:
Just curious, why would a status be not succeeded but the file would succeed?

Reply:
You're right, this shouldn't be possible. I've added a comment to FileSystem::GetStatus to mention it will always succeed for valid file descriptors.